ts/main.ts

```diff
@@ -323,6 +329,24 @@ $favoritesRow.addEventListener('click', (event: Event): void => {
         }
       } else if ($iconId === cardId && icon.classList.contains('fa-solid')) {
         icon.className = 'fa-regular fa-heart fa-2xl home-icon';
+
+        for (let i = 0; i < data.favorites.length; i++) {
+          if (data.favorites[i].id === +$iconId) {
+            data.favorites.splice(i, 1);
+          }
+        }
+
+        const $allCardElements = document.querySelectorAll('.card');
+        if (!$allCardElements) throw new Error('$allCardElements not found.');
+
+        $allCardElements.forEach((card) => {
+          if (card.getAttribute('data-id') === $iconId) {
+            const $column = card.closest('.column-fourth');
+            if (!$column) throw new Error('$column not found.');
+            $column?.remove();
+            card.remove();
+          }
```

Comment:
Do you need the optional chaining operator (`?`) on `$column`? I don't think you do because your line before it verifies that it is not `null`. Also if your `$column` is the parent of your `card`, you shouldn't have to remove the `card` too since removing the `$column` should remove all of it's children.

Reply:
I agree.  I removed the card.remove() and optional chaining operator.